src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -691,6 +691,8 @@ void QgsAppLayerTreeViewMenuProvider::editVectorSymbol()
     singleRenderer->setSymbol( symbol.release() );
     layer->triggerRepaint();
     mView->refreshLayerSymbology( layer->id() );
+    layer->emitStyleChanged();
+    QgsProject::instance()->setDirty( true );
```

Comment:
Would it make sense to connect the layer style changed signal directly to a project setDirty slot (by the project, when a layer is added)?

Reply:
That signal is fired for internal reasons too, whereas dirtying should only ever happen as a direct result of user interaction

Reply:
In this case a `void emitStyleChanged( bool makeProjectDirty = true );` might be something to consider.

Just thinking it's too easy to miss this.

Reply:
Well really having an emitXxx method in the first place is an antipattern, hinting that this whole part of the api needs a rework.

Reply:
Ok, thanks!